mint-auditor/src/gnosis/sync.rs

```diff
@@ -447,6 +468,38 @@ mod test {
                 10000000,
             ),
         ];
+
+        // Store the MintTxs in the database so we can see they get audited properly.
+        let mint_tx1 = insert_mint_tx_from_deposit(&expected_deposits[0], &conn, &mut rng);
+        let mint_tx2 = insert_mint_tx_from_deposit(&expected_deposits[1], &conn, &mut rng);
+
+        // Add another unrelated MintTx.
+        let token_id = TokenId::from(1);
+        let (_mint_config_tx, signers) = create_mint_config_tx_and_signers(token_id, &mut rng);
+        let mint_tx3 = MintTx::insert_from_core_mint_tx(
+            0,
+            None,
+            &create_mint_tx(token_id, &signers, 100, &mut rng),
+            &conn,
+        )
+        .unwrap();
+
+        // Initially, none are audited.
+        assert_eq!(
+            HashSet::<MintTx>::from_iter(vec![mint_tx1, mint_tx2, mint_tx3.clone()]),
```

Comment:
Compiler says no

```
489 |             vec![mint_tx1, mint_tx2, mint_tx3.clone()].into(),
    |             -------------------------------------------^^^^--
    |             |                                          |
    |             |                                          cannot infer type for type parameter `T` declared on the trait `Into`
    |             this method call resolves to `T`
```

```
error[E0277]: the trait bound `HashSet<mint_tx::MintTx>: From<Vec<mint_tx::MintTx>>` is not satisfied
   --> mint-auditor/src/gnosis/sync.rs:490:61
    |
490 |             MintTx::find_unaudited_mint_txs(&conn).unwrap().into(),
    |                                                             ^^^^ the trait `From<Vec<mint_tx::MintTx>>` is not implemented for `HashSet<mint_tx::MintTx>`
    |
    = help: the trait `From<[T; N]>` is implemented for `HashSet<T>`
    = note: required because of the requirements on the impl of `Into<HashSet<mint_tx::MintTx>>` for `Vec<mint_tx::MintTx>`
```

Reply:
Yeah, converting both to `into` won't work. My suggestion was converting from an array to a HashSet (note the lack of `vec!`)

Reply:
I tried each one separately

Reply:
I might've not tried without vec, I will try that.